File: Data Structures & Algorithms/two-integer-sum/submission-3.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        map<int, int> hashmap = {};

        for (int i{}; i < nums.size(); ++i) {
            if (hashmap.find(target - nums[i]) != hashmap.end()) return vector<int> {hashmap[target - nums[i]], i};
        hashmap.insert(make_pair(nums[i], i));
        }
    }
};

File: Data Structures & Algorithms/plus-one/submission-3.py
class Solution:
    def plusOne(self, digits: List[int]) -> List[int]:
        i = len(digits) - 1
        while True:
            if digits[i] != 9:
                digits[i] += 1
                return digits
            elif i != 0:
                digits[i] = 0
                i -= 1
            else:
                digits[i] = 0
                digits.insert(0, 1)
                return digits
